### PR TITLE
added opportunity to add movie images and changed username for email …

### PR DESCRIPTION
**Pull request by Timur5050**
…in user creation/login

### CODE REVIEW

File: media/upload/image/sample-movie-3d6a5884-09e7-4349-b32e-02b04517fe83.jpg
Comment (Dimosphen1) [inline media/upload/image/sample-movie-3d6a5884-09e7-4349-b32e-02b04517fe83.jpg:1]: There is no need to include files in this folder, remove them from the PR (pull request)

File: user/serializers.py [hunk 1]
```diff
@@ -22,3 +29,41 @@ def update(self, instance, validated_data):
...
+        if email and password:
+            user = authenticate(
+                request=self.context.get("request"),
+                email=email,
+                password=password
+            )
+
+            # The authenticate call simply returns None for is_active=False
+            # users. (Assuming the default ModelBackend authentication
+            # backend.)
```
Comment (Dimosphen1) [inline user/serializers.py]: There is no need to add this comment, the code is self-explicit enough, remove it

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

File: cinema/views.py [hunk 2]
```diff
@@ -101,17 +104,35 @@ def get_serializer_class(self):
...
+        methods=["POST"],
+        detail=True,
+        url_path="upload_image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
```
Comment (vsmutok) [inline cinema/views.py]: no need to use `if` here
better use `serializer.is_valid(raise_exception=True)`

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: media/upload/image/sample-movie-06e02eab-d46a-4ced-919c-889cde672797.jpg
Comment (Dimosphen1) [inline media/upload/image/sample-movie-06e02eab-d46a-4ced-919c-889cde672797.jpg:1]: Remove `.jpg` files from the PR, there is no need in them

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

**Review by vsmutok — APPROVED** [review]
Great!

**Review by DevSpaciX — APPROVED** [review]

**Pull request closed without merging**